aten/src/ATen/detail/ScalarTypeConversions.h

```diff
@@ -9,14 +9,14 @@ namespace at { namespace detail {
 
 template <typename T>
 inline T load(const void* data, ScalarType src_type) {
-  return AT_DISPATCH_ALL_TYPES_AND(at::ScalarType::Half, src_type, "load", [&]() {
+  return AT_DISPATCH_ALL_TYPES_AND2(at::ScalarType::Half, at::ScalarType::Bool, src_type, "load", [&]() {
```

Comment:
I don't understand how `store` and `load` are used, could you give me a code pointer for how TensorIterator (I'm assuming it is the user) uses these?

Reply:
it's used in Loops.cuh: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cuda/Loops.cuh#L198
to get scalar_value. And we need to be able to do this for Bool as well for functions like tensor.lt(True)